analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnModulus.cs

```diff
@@ -34,6 +35,14 @@ public sealed class EqualityOnModulus : SonarDiagnosticAnalyzer
         private const string DiagnosticId = "S2197";
         private const string MessageFormat = "The result of this modulus operation may not be {0}.";
 
+        private static readonly string[] MethodsOrPropertiesReturningAlwaysPositiveInt =
+        {
+            "System.Array.Length",
+            "System.Collections.Generic.IEnumerable<int>.Count<int>()",
+            "System.Collections.Generic.IEnumerable<int>.LongCount<int>()",
```

Comment:
Ignorable: I am thinking if we should cover `String.Length` and `List.Capacity` or not. It might be an overkill and in the same time it is an FP